lib/generators/newrelic/setup_generator.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+module Newrelic
+  class SetupGenerator < Rails::Generators::Base
+    NEWRELIC_GEM_NAME = 'newrelic_rpm'
+    NEWRELIC_GEM_USED_REGEXP = /gem.*\b#{NEWRELIC_GEM_NAME}\b.*/
+
+    desc 'This generator adds the newrelic gem and creates all the needed config files.'
+    argument :app_name, type: :string, required: true
+
+    def add_new_relic_gem
+      gem(NEWRELIC_GEM_NAME) unless newrelic_installed?
+
+      run 'bundle install'
+
+      require 'new_relic/cli/command'
+      NewRelic::Cli::Install.new(quiet: true, app_name:, license_key: "<%= ENV['NEW_RELIC_API_KEY'] %>").run
```

Comment:
after running this command I think we need to change in the generated `config/newrelic.yml`:
```
development:
  <<: *default_settings
  app_name: asd (Development)

test:
  <<: *default_settings
  # It doesn't make sense to report to New Relic from automated test runs.
  monitor_mode: false

staging:
  <<: *default_settings
  app_name: asd (Staging)
```
for
```
development:
  # It doesn't make sense to report to New Relic from the development environment.
  monitor_mode: false

test:
  <<: *default_settings
  # It doesn't make sense to report to New Relic from automated test runs.
  monitor_mode: false
```